tests/testsuite.src/run_misc.at

```diff
@@ -3617,6 +3617,111 @@ AT_CHECK([$COBCRUN_DIRECT ./prog2], [0], [], [])
+              10 sort-key-num              pic 9(04)  comp-x.
+              10 offset-to-component       pic 9(04)  comp-x.
+              10 skey-Flags                pic 9(02)  comp-x.
+              10 skey-Compression          pic 9(02)  comp-x.
+              10 sparse-Character          pic x(01).
+              10 filler                        pic x(09).
+           05 Component-Def-Area.           
+              10 sort-key            occurs max-sort-key-num.
+                 15 skey-ascending         pic x(01).
+                 15 skey-type              pic x(01).
+                 15 skey-offset            pic 9(09)  comp-x.
+                 15 skey-size              pic 9(09)  comp-x.
+       PROCEDURE        DIVISION.
+       MAIN-RTN.
+      *    INIT
+           MOVE LOW-VALUES TO FCD.
+           MOVE LOW-VALUES TO sort-key-info.
+           MOVE FCD--VERSION-NUMBER   TO FCD-VERSION
+           SET  FCD-FILENAME-ADDRESS  TO ADDRESS OF EX-FILENAME.
+           MOVE 80                    TO FCD-NAME-LENGTH.
+           SET FCD-KEY-DEF-ADDRESS TO
+                             ADDRESS OF sort-key-info.
+           MOVE SD-init               TO OPCODE.
+           SET FCD-RECORD-ADDRESS     TO ADDRESS OF EX-RECORD-BUFFER.
+           MOVE 10                    TO FCD-CURRENT-REC-LEN.
+           MOVE 10                    TO FCD-MIN-REC-LENGTH.
+           MOVE 10                    TO FCD-MAX-REC-LENGTH.
+           MOVE 1                     TO sort-key-num.
+           move 1                     TO sort-key-count.
+           MOVE 0                     TO skey-offset(1).
+           MOVE 3                     TO skey-size(1).
+           move 30                    TO offset-to-component.
+           move X"80"                 TO skey-ascending(1).
+           move X"00"                 TO skey-type(1).
+      *>   "INIT".
+           CALL "EXTSM" USING OPCODE, FCD.
+      *>   "RELEASE".
+           MOVE SD-release TO OPCODE.
+           MOVE "BBB111" TO EX-RECORD-BUFFER.
+           CALL "EXTSM" USING OPCODE, FCD.
+
+           MOVE "CCC333" TO EX-RECORD-BUFFER.
+           CALL "EXTSM" USING OPCODE, FCD.
+
+           MOVE "AAA222" TO EX-RECORD-BUFFER.
+           CALL "EXTSM" USING OPCODE, FCD.
+      *>   "INPUT END".
+           MOVE SD-inp-end TO OPCODE.
+           CALL "EXTSM" USING OPCODE, FCD.
+
+      *>   "RETURN".
+           MOVE SD-return TO OPCODE.
+           CALL "EXTSM" USING OPCODE, FCD.
+           DISPLAY EX-RECORD-BUFFER.
+
+           CALL "EXTSM" USING OPCODE, FCD.
+           DISPLAY EX-RECORD-BUFFER.
+
+           CALL "EXTSM" USING OPCODE, FCD.
+           DISPLAY EX-RECORD-BUFFER.
+
+      *>   "TERM".
+           MOVE SD-term TO OPCODE.
+           CALL "EXTSM" USING OPCODE, FCD.
+       MAIN-EXT.
+           STOP RUN.
+])
+
+AT_CHECK([$COMPILE -std=mf prog.cob], [0], [], [])
+AT_CHECK([$COBCRUN_DIRECT ./prog], [0],
+[AAA222
+BBB111
+CCC333
+], [])
+
+AT_CLEANUP
+
```

Comment:
This would be OK for an initial test. I do wonder if you have more EXTSM testcases, though.